src/components/Chart/convertSeries.ts

```diff
@@ -20,20 +21,59 @@ export function convertSeries<XDomain>(
     })
   })
 
+  const addTick = (x: XDomain): XDomain => {
+    if (typeof x === 'number') return ((x + 1) as unknown) as XDomain
+    if (x instanceof Date) return (new Date(x.getTime() + 1) as unknown) as XDomain
+    if (typeof x === 'object' && x !== null) return ((+x + 1) as unknown) as XDomain
+    return x
+  }
+
+  const hiddenRanges = rangeAreas?.filter((r) => r.mask?.show && !r.mask?.showDots) ?? []
```

Comment:
I think you can pass only this as parameter. This method does not need to know the entire `rangeAreas`

Reply:
At line 66, I now use the range series to collect all range series with yAtEnd set. From now on, I’ll be passing two props: hiddenRanges and rangesWithYAtEnd.